### ISSUE TITLE
Can't compile version 24.08.2 or 24.05 on Debian 12 64bit

### ISSUE THREAD

**Issue by tioneb22** [ElementsProject/lightning#7752]
### Issue and Steps to Reproduce
I'm following the RaspiBolt [guide](https://raspibolt.org/guide/bonus/lightning/cln.html#introduction) to build CLN from scratch; the guide is for version 23.08.1, but I'd thought I'd try my luck with the new version.

I'm running into this error when running make:

`make: *** [Makefile:394: contrib/pyln-grpc-proto/pyln/grpc/primitives_pb2.py] Error 1
`

I've installed these extra libraries (which weren't in the guide) to try and overcome this issue, but it still won't compile:
`jq libffi-dev python3-grpc-tools protobuf-compiler
`

Moreover, I was curious so I checked out version 23.08.1 (as the guide suggested) and it compiled fine. However, I couldn't start the daemon when I had TOR settings enabled in the config. Disabling the TOR settings (proxy, bind-addr, addr, always-use-proxy) allowed me to start the daemon successfully. Here is the screenshot of daemon failing to start because of TOR.

![Screenshot from 2024-10-19 00-06-14](https://github.com/user-attachments/assets/7fd1e198-a5a8-4842-b3a9-b9e118d4c536)

![Screenshot from 2024-10-19 00-15-05](https://github.com/user-attachments/assets/e6fb8fe2-fd25-4d09-be4b-288398a6fba7)

**Comment by tioneb22** [comment]
I can't compile v24.02 or v24.02.2 either, getting same error:
![Screenshot from 2024-11-01 21-36-28](https://github.com/user-attachments/assets/24a5bf3e-0d76-4a02-8966-dab9d367a8b2)

Does anyone have any suggestions on what I can try?

**Comment by tioneb22** [comment]
I figured it out!

My CLN user was not using the system version of grpc; it was using version 1.14.0.
The "--experimental_allow_proto3_optional" flag was released in version 1.38.0.
Th fix was to upgrade the grpc package as the CLN user using pip:
`
pip3 install --upgrade grpcio-tools --break-system-packages
`
It wouldn't allow me to upgrade the package without the "--break-system-packages" flag... it was already broken so I ignored that lol.

I have now compiled version 24.08.2 and I'm up and running!
